packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/Convert.java

```diff
@@ -690,103 +690,36 @@ private static void interpretInfoWindowOptions(
-    } else {
-      return polygonId;
-    }
+  static String interpretPolygonOptions(Messages.PlatformPolygon polygon, PolygonOptionsSink sink) {
+    sink.setConsumeTapEvents(polygon.getConsumesTapEvents());
+    sink.setGeodesic(polygon.getGeodesic());
+    sink.setVisible(polygon.getVisible());
+    sink.setFillColor(polygon.getFillColor().intValue());
+    sink.setStrokeColor(polygon.getStrokeColor().intValue());
+    sink.setStrokeWidth(polygon.getStrokeWidth());
+    sink.setZIndex(polygon.getZIndex());
+    sink.setPoints(toPoints(polygon.getPoints()));
+    sink.setHoles(toHoles(polygon.getHoles()));
+    return polygon.getPolygonId();
   }
 
   static String interpretPolylineOptions(
-      Map<String, ?> data, PolylineOptionsSink sink, AssetManager assetManager, float density) {
-    final Object consumeTapEvents = data.get("consumeTapEvents");
-    if (consumeTapEvents != null) {
-      sink.setConsumeTapEvents(toBoolean(consumeTapEvents));
-    }
-    final Object color = data.get("color");
-    if (color != null) {
-      sink.setColor(toInt(color));
-    }
-    final Object endCap = data.get("endCap");
-    if (endCap != null) {
-      sink.setEndCap(toCap(endCap, assetManager, density));
-    }
-    final Object geodesic = data.get("geodesic");
-    if (geodesic != null) {
-      sink.setGeodesic(toBoolean(geodesic));
-    }
-    final Object jointType = data.get("jointType");
-    if (jointType != null) {
-      sink.setJointType(toInt(jointType));
-    }
-    final Object startCap = data.get("startCap");
-    if (startCap != null) {
-      sink.setStartCap(toCap(startCap, assetManager, density));
-    }
-    final Object visible = data.get("visible");
-    if (visible != null) {
-      sink.setVisible(toBoolean(visible));
-    }
-    final Object width = data.get("width");
-    if (width != null) {
-      sink.setWidth(toInt(width));
-    }
-    final Object zIndex = data.get("zIndex");
-    if (zIndex != null) {
-      sink.setZIndex(toFloat(zIndex));
-    }
-    final Object points = data.get("points");
-    if (points != null) {
-      sink.setPoints(toPoints(points));
-    }
-    final Object pattern = data.get("pattern");
-    if (pattern != null) {
-      sink.setPattern(toPattern(pattern));
-    }
-    final String polylineId = (String) data.get("polylineId");
-    if (polylineId == null) {
-      throw new IllegalArgumentException("polylineId was null");
-    } else {
-      return polylineId;
-    }
+      Messages.PlatformPolyline polyline,
+      PolylineOptionsSink sink,
+      AssetManager assetManager,
+      float density) {
+    sink.setConsumeTapEvents(polyline.getConsumesTapEvents());
+    sink.setColor(polyline.getColor().intValue());
+    sink.setEndCap(toCap(polyline.getEndCap(), assetManager, density));
+    sink.setStartCap(toCap(polyline.getStartCap(), assetManager, density));
+    sink.setGeodesic(polyline.getGeodesic());
+    sink.setJointType(polyline.getJointType().intValue());
```

Comment:
This is fine for this PR, since everything here is a net improvement, but please add a TODO to the `jointType` field in `messages.dart` to convert it to an enum. Currently the way this is working is that the Dart definition of the Dart enum is explicitly assigning numerical values that match the *native* SDK's constant value, which is still super fragile. Even though it's more code, the right way to handle that is to convert on the Dart side from the platform interface enum to a Pigeon-defined enum, and then convert on the Java side from the Pigeon enum to the native SDK constants, so that the raw int values used in the SDK don't matter and don't have to match things in other packages.

Reply:
The TODO can reference https://github.com/flutter/flutter/issues/153718.